app/components/PersonNameField.tsx

```diff
@@ -0,0 +1,154 @@
+import { useCallback } from "react";
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+import { useForm } from "react-hook-form";
+import { Person } from "@/types/person";
+import { useMutation } from "@/utils/graphql";
+
+interface PersonNameFieldProps {
+  /**
+   * Record corresponding to row being edited
+   */
+  record: Person;
+  /**
+   * Name of record mutation
+   */
+  mutation: string;
+  /**
+   * The function to call on cancel button click, which is expected
+   * to set the parent component's edit state
+   */
+  onCancel: () => void;
+  /**
+   * Function that is an async wrapper around data refetch
+   */
+  onSaveCallback: () => Promise<void>;
+  /**
+   * This column's edit state
+   */
+  isEditingThisColumn: boolean;
+}
+
+type PersonNameFormInputs = {
+  first_name: string | null;
+  middle_name: string | null;
+  last_name: string | null;
```

Comment:
might as well prefix all these fieldnames with `prsn_` so that you don't have to modify them when you do the mutation?

i guess also because we're going with the snake casing here i am expecting these fields to match a db fieldname exactly 🪞